src/urllib3/response.py

```diff
@@ -827,6 +827,8 @@ def _fp_read(
                 del data  # to reduce peak memory usage by `max_chunk_amt`.
             return buffer.getvalue()
         elif read1:
+            # work around: https://github.com/python/cpython/issues/112064
+            amt = amt or self.length_remaining
             return self._fp.read1(amt) if amt is not None else self._fp.read1()
```

Comment:
I don't see why this is needed. The mentioned issue is related to passing `-1` to `HTTPResponse.read`, this part of code doesn't use the method.

Reply:
Not sure what I was thinking there, sorry, reverted!